src/NLog/Internal/UrlHelper.cs

```diff
@@ -33,82 +33,152 @@
 
 namespace NLog.Internal
 {
+    using System;
     using System.Text;
 
     /// <summary>
     /// URL Encoding helper.
     /// </summary>
-    internal class UrlHelper
+    internal static class UrlHelper
     {
-        private static string safeUrlPunctuation = ".()*-_!'";
-        private static string hexChars = "0123456789abcdef";
+        private static readonly char[] hexUpperChars =
+            { '0', '1', '2', '3', '4', '5', '6', '7',
+            '8', '9', 'A', 'B', 'C', 'D', 'E', 'F' };
+        private static readonly char[] hexLowerChars =
+            { '0', '1', '2', '3', '4', '5', '6', '7',
+            '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
+
+        [Flags]
+        public enum EscapeEncodingFlag
+        {
+            /// <summary>Use UnreservedMarks instead of ReservedMarks, as specified by chosen RFC</summary>
+            UriString = 1,
+            /// <summary>Use RFC2396 standard (instead of RFC3986)</summary>
+            LegacyRfc2396 = 2,
+            /// <summary>Should use lowercase when doing HEX escaping of special characters</summary>
+            LowerCaseHex = 4,
+            /// <summary>Replace space ' ' with '+' instead of '%20'</summary>
+            SpaceAsPlus = 8,
+            /// <summary>Skip UTF8 encoding, and prefix special characters with '%u'</summary>
+            NLogLegacy = 16 | LegacyRfc2396 | LowerCaseHex | UriString,
+        };
+
+        private const string RFC2396ReservedMarks = @";/?:@&=+$,";
+        private const string RFC3986ReservedMarks = @":/?#[]@!$&'()*+,;=";
+        private const string RFC2396UnreservedMarks = @"-_.!~*'()";
+        private const string RFC3986UnreservedMarks = @"-._~";
 
         /// <summary>
-        /// Url encode and URL
+        /// Url encode an URL
         /// </summary>
         /// <param name="str">URL to be encoded</param>
         /// <param name="spaceAsPlus">space as + or %20? <c>false</c> (%20) is the safe option.</param>
         /// <returns>Encoded url.</returns>
-        internal static string UrlEncode(string str, bool spaceAsPlus)
+        public static string UrlEncode(string str, bool spaceAsPlus)
```

Comment:
Well it is still hidden by the internal class protection, and it is meant to be used outside this class